src/main/scala/org/constellation/infrastructure/endpoints/SnapshotEndpoints.scala

```diff
@@ -116,18 +119,33 @@ class SnapshotEndpoints[F[_]](implicit F: Concurrent[F], C: ContextShift[F]) ext
         redownloadService.getAcceptedSnapshots().map(_.asJson).flatMap(Ok(_))
     }
 
-  private def getPeerProposals(nodeId: Id, redownloadService: RedownloadService[F]): HttpRoutes[F] =
+  private def getPeerProposals(selfId: Id, redownloadService: RedownloadService[F]): HttpRoutes[F] =
     HttpRoutes.of[F] {
       case GET -> Root / "peer" / peerId / "snapshot" / "created" =>
         val peerProposals =
-          if (Id(peerId) == nodeId)
+          if (Id(peerId) == selfId)
             redownloadService.getCreatedSnapshots()
           else
-            redownloadService.getPeerProposals(nodeId).map(_.getOrElse(Map.empty))
+            redownloadService.getPeerProposals(Id(peerId)).map(_.getOrElse(Map.empty))
```

Comment:
You could create Id(peerId) once and then reuse it.

Reply:
I hoped that the compiler or jit will do that for me. But you might be right, it may not.